ReactiveCocoa/Swift/Property.swift

```diff
@@ -8,66 +8,457 @@ public protocol PropertyType {
+	return zip(a, b, c, d, e)
+		.zip(with: f)
+		.map(repack)
+}
+
+/// Zips the values of all the given properties, in the manner described by
+/// `zip(with:)`.
+@warn_unused_result(message="Did you forget to use the property?")
+public func zip<A: PropertyType, B: PropertyType, C: PropertyType, D: PropertyType, E: PropertyType, F: PropertyType, G: PropertyType>(a: A, _ b: B, _ c: C, _ d: D, _ e: E, _ f: F, _ g: G) -> AnyProperty<(A.Value, B.Value, C.Value, D.Value, E.Value, F.Value, G.Value)> {
+	return zip(a, b, c, d, e, f)
+		.zip(with: g)
+		.map(repack)
+}
+
+/// Zips the values of all the given properties, in the manner described by
+/// `zip(with:)`.
+@warn_unused_result(message="Did you forget to use the property?")
+public func zip<A: PropertyType, B: PropertyType, C: PropertyType, D: PropertyType, E: PropertyType, F: PropertyType, G: PropertyType, H: PropertyType>(a: A, _ b: B, _ c: C, _ d: D, _ e: E, _ f: F, _ g: G, _ h: H) -> AnyProperty<(A.Value, B.Value, C.Value, D.Value, E.Value, F.Value, G.Value, H.Value)> {
+	return zip(a, b, c, d, e, f, g)
+		.zip(with: h)
+		.map(repack)
+}
+
+/// Zips the values of all the given properties, in the manner described by
+/// `zip(with:)`.
+@warn_unused_result(message="Did you forget to use the property?")
+public func zip<A: PropertyType, B: PropertyType, C: PropertyType, D: PropertyType, E: PropertyType, F: PropertyType, G: PropertyType, H: PropertyType, I: PropertyType>(a: A, _ b: B, _ c: C, _ d: D, _ e: E, _ f: F, _ g: G, _ h: H, _ i: I) -> AnyProperty<(A.Value, B.Value, C.Value, D.Value, E.Value, F.Value, G.Value, H.Value, I.Value)> {
+	return zip(a, b, c, d, e, f, g, h)
+		.zip(with: i)
+		.map(repack)
+}
+
+/// Zips the values of all the given properties, in the manner described by
+/// `zip(with:)`.
+@warn_unused_result(message="Did you forget to use the property?")
+public func zip<A: PropertyType, B: PropertyType, C: PropertyType, D: PropertyType, E: PropertyType, F: PropertyType, G: PropertyType, H: PropertyType, I: PropertyType, J: PropertyType>(a: A, _ b: B, _ c: C, _ d: D, _ e: E, _ f: F, _ g: G, _ h: H, _ i: I, _ j: J) -> AnyProperty<(A.Value, B.Value, C.Value, D.Value, E.Value, F.Value, G.Value, H.Value, I.Value, J.Value)> {
+	return zip(a, b, c, d, e, f, g, h, i)
+		.zip(with: j)
+		.map(repack)
+}
+
+/// Zips the values of all the given properties, in the manner described by
+/// `zip(with:)`. Returns nil if the sequence is empty.
+@warn_unused_result(message="Did you forget to call `start` on the producer?")
+public func zip<S: SequenceType where S.Generator.Element: PropertyType>(properties: S) -> AnyProperty<[S.Generator.Element.Value]>? {
+	var generator = properties.generate()
+	if let first = generator.next() {
+		let initial = first.map { [$0] }
+		return GeneratorSequence(generator).reduce(initial) { property, next in
+			property.zip(with: next).map { $0.0 + [$0.1] }
+		}
+	}
+
+	return nil
+}
+
+/// A read-only, type-erased view of a property.
```

Comment:
It erased the type of what actually `PropertyType` is.

https://github.com/apple/swift/blob/master/stdlib/public/core/ExistentialCollection.swift.gyb